lighthouse-plots/analyze.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const utils = require('./utils');
+
+const OUT_PATH = path.resolve(__dirname, 'out');
+const GENERATED_RESULTS_PATH = path.resolve(OUT_PATH, 'generatedResults.js');
+const LIGHTHOUSE_RESULTS_FILENAME = 'lighthouse.json';
+
+const METRICS = [
+  'firstMeaningfulPaint',
+  'speedIndexMetric',
+  'estimatedInputLatency',
+  'timeToInteractive',
+  'visuallyReady',
+  'traceEnd'
+];
+
+// TODO: annotate
+function main() {
+  const allResults = [];
+  fs.readdirSync(OUT_PATH).map(siteDir => {
+    const sitePath = path.resolve(OUT_PATH, siteDir);
+    if (!utils.isDir(sitePath)) {
+      return;
+    }
+    allResults.push({ site: siteDir, results: analyzeSite(sitePath) });
+  });
+  const formattedResults = formatResults(allResults);
+  fs.writeFileSync(
+    GENERATED_RESULTS_PATH,
+    `var generatedResults = ${JSON.stringify(formattedResults)}`
+  );
+}
+
+main();
+
+/**
+ * @param {string} sitePath
+ * @return {!RunResults}
+ */
+function analyzeSite(sitePath) {
+  console.log('Analyzing', sitePath);
+  const siteResults = [];
+  fs.readdirSync(sitePath).forEach(runDir => {
+    const runPath = path.resolve(sitePath, runDir);
+    const lighthouseReportPath = path.resolve(runPath, 'lighthouse.json');
+    if (!utils.isDir(runPath) || !utils.isFile(lighthouseReportPath)) return;
+    const metrics = readResult(runPath);
+    console.log(`Metric for ${runDir}: ${JSON.stringify(metrics)}`);
+    siteResults[runDir] = {
+      runId: runDir,
+      metrics
+    };
+  });
+  return siteResults;
+}
+
+/**
+ * @param {string} resultPath
+ * @return {!Metrics}
+ */
+function readResult(resultPath) {
+  const data = JSON.parse(
+    fs.readFileSync(path.resolve(resultPath, LIGHTHOUSE_RESULTS_FILENAME))
+  );
+  const audits = data.audits;
+  return {
+    firstMeaningfulPaint: {
```

Comment:
I'm thinking it'd be advantageous to reuse the definitions inside of [pwmetrics-events.js](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/lib/traces/pwmetrics-events.js#L31). They are primarily used for adding the LH metrics back into the trace, but the external `pwmetrics` module also depends on it.

As a usecase, @patrickhulce is adding some new TTI values now and we'll want to have them both embedded in the trace and summarized here in plots.  

I didn't capture score over there, but I'm okay adding it. I'm also OK with ignoring score completely for now.